src/coreclr/jit/gentree.h

```diff
@@ -118,16 +118,23 @@ enum gtCallTypes : BYTE
 
 enum class ExceptionSetFlags : uint32_t
 {
-    None                     = 0x0,
-    OverflowException        = 0x1,
-    DivideByZeroException    = 0x2,
-    ArithmeticException      = 0x4,
-    NullReferenceException   = 0x8,
-    IndexOutOfRangeException = 0x10,
-    StackOverflowException   = 0x20,
+    None                        = 0x0,
+    OverflowException           = 0x1,
+    DivideByZeroException       = 0x2,
+    ArithmeticException         = 0x4,
+    NullReferenceException      = 0x8,
+    IndexOutOfRangeException    = 0x10,
+    StackOverflowException      = 0x20,
+#ifdef TARGET_ARM64
+    ArgumentOutOfRangeException = 0x40,
```

Comment:
Notably was talking about this with @jakobbotsch on Discord earlier and mentioned that we can't model `ArgumentOutOfRangeException` with `OperExceptions` because we can't reorder two such exceptions due to containing parameters.

I don't have the context as to why that is, but my guess is there's some other usage/handling of these that makes it problematic

Reply:
What should we do instead?